assets/app/scripts/directives/truncate.js

```diff
@@ -3,34 +3,21 @@
 angular.module('openshiftConsole')
   // Truncates text to a length, adding a tooltip and an ellipsis if truncated.
   // Different than `text-overflow: ellipsis` because it allows for multiline text.
-  .directive('truncateLongText', function() {
+  .directive('truncateLongText', function(truncateFilter) {
     return {
       restrict: 'E',
       scope: {
         content: '=',
         limit: '=',
+        newlineLimit: '=',
         useWordBoundary: '='
       },
       template: '<span ng-attr-title="{{content}}">{{visibleContent}}<span ng-if="truncated">&hellip;</span></span>',
       link: function(scope) {
-        scope.visibleContent = scope.content;
         scope.$watch('content', function(content) {
-          if (!scope.limit || !content || content.length <= scope.limit) {
-            scope.truncated = false;
-            scope.visibleContent = content;
-            return;
-          }
-
-          scope.truncated = true;
-          scope.visibleContent = content.substring(0, scope.limit);
-          if (scope.useWordBoundary !== false) {
-            // Find the last word break, but don't look more than 10 characters back.
-            // Make sure we show at least the first 5 characters.
-            var startIndex = Math.max(4, scope.limit - 10);
-            var lastSpace = scope.visibleContent.lastIndexOf(' ', startIndex);
-            if (lastSpace !== -1) {
-              scope.visibleContent = scope.visibleContent.substring(0, lastSpace);
-            }
+          scope.visibleContent = truncateFilter(content, scope.limit, scope.useWordBoundary, scope.newlineLimit);
+          if (scope.visibleContent.length !== content.length) {
+            scope.truncated = true;
```

Comment:
If the content changes to something short, this is never reset to false. Maybe just

```
scope.truncated = scope.visibleContent.length !== content.length;
```

or you could even just check that in the view and get rid of the scope var.